### PR TITLE
[Pages] Simplify Next.js docs (and port over content from `next-on-pages` repo)

### PR DESCRIPTION
**Pull request by kodster28**
### Summary

An attempt to address the comment at https://github.com/cloudflare/next-on-pages/pull/850#pullrequestreview-2219385438, where we provide a streamlined guide (as suggested in #15971), but then do the following:

- Create a new sub-section for `Fullstack (SSR)`
- Moves binding and troubleshooting content from the `get started` guide onto their own pages
- Starts migration of existing info from the `next-on-pages` repo, with `caching` as the first example page.

Might be slightly misunderstanding how those `next-on-pages` docs are scoped... my assumption here is that they're mostly relevant to SSR vs static, but let me know if that's an incorrect assumption.

Branched off #15971 since it was a good start on the consolidation.

### CODE REVIEW

File: content/pages/framework-guides/nextjs/ssr/caching.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+pcx_content_type: reference
+title: Caching
+meta:
+  title: Caching and data revalidation in your Next.js app
+---
+
+
+# Caching and data revalidation
```
Comment (irvinebroque) [inline content/pages/framework-guides/nextjs/ssr/caching.md:8]: Have questions about this section, see internal chat, would like to resolve them and get clear before we put page on this in docs.

File: content/pages/framework-guides/nextjs/ssr/caching.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+When you write cached data to Workers KV, you write to storage that can be read by any Cloudflare location. This means your app can fetch data, cache it in KV, and then subsequent requests anywhere around the world can read from this cache.
+
+Note that Workers KV is eventually consistent, which means that it can take up to 60 seconds for updates to be reflected globally.
+
+To use Workers KV as the cache for your Next.js app, [add a KV binding](/pages/functions/bindings/#kv-namespaces) to your Pages project, and set the name of the binding to `__NEXT_ON_PAGES__KV_SUSPENSE_CACHE`.
+
+### Cache API (default)
+
+The [Cache API](https://developers.cloudflare.com/workers/runtime-apis/cache/) is the default option for caching data in your Next.js app. You do not need to take any action to enable the Cache API.
```
[the remark this answers is not in the document]
Comment (irvinebroque) [reply content/pages/framework-guides/nextjs/ssr/caching.md:36]:
```suggestion
The [Cache API](/workers/runtime-apis/cache/) is the default option for caching data in your Next.js app. You do not need to take any action to enable the Cache API.
```

File: content/pages/framework-guides/nextjs/ssr/get-started.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  description: Deploy a full-stack Next.js app to Cloudflare Pages
+---
+
+# Next.js
+
+Learn how to deploy your Full-stack (SSR) Next.js on Cloudflare.
+
+## New apps
+
+For new Next.js apps, use the following command to launch an interactive building experience:
```
Comment (irvinebroque) [inline content/pages/framework-guides/nextjs/ssr/get-started.md]:
```suggestion
To create a new Next.js app, pre-configured to run on Cloudflare, run:
```

Just cut to the chase, fewer words better.

File: content/pages/framework-guides/nextjs/ssr/get-started.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+title: Get started
+weight: 1
+meta:
+  title: Get started | Full-stack (SSR) | Next.js apps
+  description: Deploy a full-stack Next.js app to Cloudflare Pages
+---
+
+# Next.js
+
+Learn how to deploy your Full-stack (SSR) Next.js on Cloudflare.
```
Comment (irvinebroque) [inline content/pages/framework-guides/nextjs/ssr/get-started.md]:
```suggestion
Learn how to deploy full-stack (SSR) Next.js apps to Cloudflare Pages.
```

File: content/pages/framework-guides/nextjs/ssr/get-started.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+$ npm create cloudflare@latest my-next-app -- --framework=next
+```
+
+For more guidance on developing your app, refer to [Bindings](/pages/framework-guides/nextjs/ssr/bindings/) or the [Next.js documentation](https://nextjs.org).
+
+---
+
+## Existing apps
+
+For existing apps, you need to perform a few more steps.
```
Comment (irvinebroque) [inline content/pages/framework-guides/nextjs/ssr/get-started.md]:
```suggestion
```

File: content/pages/framework-guides/nextjs/ssr/_index.md
Comment (james-elicx) [inline content/pages/framework-guides/nextjs/ssr/_index.md:1]: can you add back the advanced usage section you're removing in next-on-pages too?

Comment (irvinebroque) [reply content/pages/framework-guides/nextjs/ssr/_index.md:1]: https://github.com/cloudflare/cloudflare-docs/commit/0257fe513d3744e5d4fd655b228ab058f617fdc1

File: content/pages/framework-guides/nextjs/ssr/advanced.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+} as ExportedHandler<{ ASSETS: Fetcher }>;
+```
+
+This looks like a Worker — but it does not need its own `wrangler.toml` file. You can think of it purely as code that `@cloudflare/next-on-pages` will then use to wrap the output of the build that is deployed to your Cloudflare Pages project.
+
+
+2. Pass the entrypoint argument to the next-on-pages CLI with the path to your handler.
+
+```sh
+npx @cloudflare/next-on-pages --custom-entrypoint=./custom-entrypoint.ts
```
Comment (irvinebroque) [inline content/pages/framework-guides/nextjs/ssr/advanced.md]:
```suggestion
$ npx @cloudflare/next-on-pages --custom-entrypoint=./custom-entrypoint.ts
```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+# Supported Versions and Features
+
+## Supported Next.js versions
+
+`@cloudflare/next-on-pages` supports all minor and patch version of Next.js 13 and 14. We regularly run manual and automated tests to ensure compatibility.
+
+### Node.js
+
+Next.js has [two "runtimes"](https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes) — "Edge" and "Node.js". When you run your Next.js app on Cloudflare, you [can use available Node.js APIs](/workers/runtime-apis/nodejs/) — but you currently can only use Next.js' "Edge" runtime.
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md:16]: I might reframe for clarity here by moving the limitations to the front, then providing the explanation.

Comment (kodster28) [reply content/pages/framework-guides/nextjs/ssr/supported-features.md:16]: So it'd read more like, 

```
The following Node.js APIs work in a Next.js app:

<list>

Explanation as to why that is.
```

Comment (kodster28) [reply content/pages/framework-guides/nextjs/ssr/supported-features.md:16]: Happy to push up that change if it makes sense

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+- `events`
+- `assert`
+- `util`
+- `async_hooks`
+
+## Supported Features
+
+### Routers
+
+Both the older [Pages](https://nextjs.org/docs/pages) router and newer [App](https://nextjs.org/docs/app) router from Next.js are supported. Cloudlflare recommends using the App router.
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md]:
```suggestion
Cloudlflare recommends using the [App router](https://nextjs.org/docs/app) from Next.js.

Cloudflare also supports the older [Pages](https://nextjs.org/docs/pages) router from Next.js.
```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+[^11]: **reactStrictMode**: Currently we build the application so react strict mode (being a local dev feature) doesn't work either way. If we can make strict mode work, this option will most likely work straight away.
+
+[^12]: **runtime configuration**: We could look into implementing the runtime configuration but it is probably not worth it since it is a legacy configuration and environment variables should be used instead.
+
+[^13]: **serverComponentsExternalPackages**: This option is for applications running on Node.js so it's not relevant to applications running on Cloudflare Pages.
+
+### Internationalization
+
+Besides the above mentioned `next.config.js` properties, there is also the `i18n` one, that is also fully supported meaning that `@cloudflare/next-on-pages` does support Next.js' built-in internationalization system. For more details on the option see the [Next.js Internationalization documentation](https://nextjs.org/docs/pages/building-your-application/routing/internationalization).
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md]:
```suggestion
Cloudflare also supports Next.js' [internationalized (`i18n`) routing](https://nextjs.org/docs/pages/building-your-application/routing/internationalization).
```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+ISR pages are built by the Vercel CLI to generate Vercel [Prerender Functions](https://vercel.com/docs/build-output-api/v3/primitives#prerender-functions). These are Node.js serverless functions that can be called in the background while serving the page from the cache. It is not possible to use these with Cloudflare Pages and they are not compatible with the [edge runtime](https://nextjs.org/docs/app/api-reference/edge) currently.
+
+In case the Vercel build process generates prerendered pages for your application, `@cloudflare/next-on-pages` will use static fallback files that are generated by the build process so that your application will still correctly serve your ISR/prerendered pages (but without the regeneration aspect).
+
+#### Dynamic handling of static routes
+
+`@cloudflare/next-on-pages` supports standard statically generated routes, it does however not support dynamic Node.js-based on-demand handling of such routes.
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md]:
```suggestion
`@cloudflare/next-on-pages` supports standard statically generated routes. 

 It does not support dynamic Node.js-based on-demand handling of such routes.
```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+For more details see:
+
+- [troubleshooting `generateStaticParams`](/pages/framework-guides/nextjs/ssr/troubleshooting/#generatestaticparams)
+- [troubleshooting `getStaticPaths` ](/pages/framework-guides/nextjs/ssr/troubleshooting/#getstaticpaths)
+
+#### Caching and Data Revalidation
+
+Revalidation and `next/cache` are supported on Cloudflare Pages, and can use various bindings. For more information, see our [caching documentation](/pages/framework-guides/nextjs/ssr/caching/).
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md]:
```suggestion
Revalidation and `next/cache` are supported on Cloudflare Pages and can use various bindings. For more information, see our [caching documentation](/pages/framework-guides/nextjs/ssr/caching/).
```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+
+### Internationalization
+
+Besides the above mentioned `next.config.js` properties, there is also the `i18n` one, that is also fully supported meaning that `@cloudflare/next-on-pages` does support Next.js' built-in internationalization system. For more details on the option see the [Next.js Internationalization documentation](https://nextjs.org/docs/pages/building-your-application/routing/internationalization).
+
+### Rendering and Data Fetching
+
+#### Incremental Static Regeneration
+
+Incremental Static Regeneration (ISR) is a rendering mode in Next.js that allows you to automatically cache and periodically regenerate pages with fresh data. Next.js [does not support](https://nextjs.org/docs/pages/building-your-application/rendering/incremental-static-regeneration) building ISR pages for the edge runtime, and as such, pages should be changed to use server side rendering (SSR) instead.
```
Comment (kodster28) [inline content/pages/framework-guides/nextjs/ssr/supported-features.md]: I'd rephrase this section to bring forward the important info, something like,

```
If you use ISR pages (insert definition in footnote), `@cloudflare/next-on-pages` will use static fallback files that are generated by the build process.

This means that your application will still correctly serve your ISR/prerendered pages (but without the regeneration aspect).

If this causes issues for your application, change your pages to use server side rendering (SSR) instead. 

<insert explanation as to why it doesn't work in a dropdown>

```

File: content/pages/framework-guides/nextjs/ssr/supported-features.md (hunk 7 above)
Comment (irvinebroque) [reply content/pages/framework-guides/nextjs/ssr/supported-features.md:16]: Hmmm I would like to keep this as-is so that we can be clear about what is driving the constraint upfront, since it is an unusual one.

File: content/pages/framework-guides/nextjs/ssr/supported-features.md (hunk 12 above)
Comment (irvinebroque) [reply content/pages/framework-guides/nextjs/ssr/supported-features.md]: go for it

**Review by dcpena — APPROVED** [review]

**Review by irvinebroque — APPROVED** [review]
